pkg/aws/pca_test.go

```diff
@@ -383,6 +388,53 @@ func TestPCASign(t *testing.T) {
 	}
 }
 
+func TestPCASign(t *testing.T) {
+	type testCase struct {
+		provisioner   PCAProvisioner
+		expectFailure bool
+		expectedChain string
+		expectedCert  string
+	}
+
+	tests := map[string]testCase{
+		"success": {
+			provisioner:   PCAProvisioner{arn: arn, pcaClient: &workingACMPCAClient{}},
+			expectFailure: false,
+		},
+		"failure-error-issueCertificate": {
+			provisioner:   PCAProvisioner{arn: arn, pcaClient: &errorACMPCAClient{}},
+			expectFailure: true,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			key, _ := rsa.GenerateKey(rand.Reader, 2048)
+			csrBytes, _ := x509.CreateCertificateRequest(rand.Reader, &template, key)
+
+			cr := &v1.CertificateRequest{
+				Spec: v1.CertificateRequestSpec{
+					Request: pem.EncodeToMemory(&pem.Block{
+						Bytes: csrBytes,
+						Type:  "CERTIFICATE REQUEST",
+					}),
+				},
+			}
+
+			err := tc.provisioner.Sign(context.TODO(), cr, logr.Discard())
+
+			if tc.expectFailure && err == nil {
+				fmt.Print(err.Error())
+				assert.Fail(t, "Expected an error but received none")
+			}
+
+			if tc.expectedChain != "" && tc.expectedCert != "" {
+				assert.Equal(t, cr.ObjectMeta.GetAnnotations()["aws-privateca-issuer/certificate-arn"], certArn)
+			}
```

Comment:
This isn't going to get hit because your test cases aren't passing in an expectedChain or expectedCert. I wouldn't expect them to either since Sign is now just calling `issueCertificate`

Consider removing the `if` and any case that it isn't expecting a failure, should verify the existence of the cert arn